magma/lib/magma.rb

```diff
@@ -150,7 +150,9 @@ def server_pid
     else
       # Oh boy.  Not ideal, but best effort here.  This could end up just restarting the wrong puma process if
       # it was hosted together.  Ideally we're running these processes in separate containers so it should be ok.
-      `pidof puma | tail -n 1`.chomp.to_i
```

Comment:
So, firstly, the pidfile approach isn't reliable -- I'm not actually sure that the pidfile is working in containerization altogether, so this branch is the one that gets hit.
Previously, `pidof puma` fails because the actual process is `ruby`, running the `puma` script.
Fortunately `pgrep` is much better and finds the process.  Better yet, we now return nil (rather than 0) when we can't find the process, which means we'll catch these problems in the future due to the kill command raising an exception in that case.

The result of the previous command was `"".to_i` => `0` => kills the entire process group, forcing swarm or docker compose to restart the entire container as opposed to merely, reboot the puma process, which is orders of magnitude faster.